### ISSUE TITLE
cannot open input file 'BWAPI.lib'

### ISSUE THREAD

**Issue by TheGeekiestOne** [davechurchill/commandcenter#86]
No lib folder in BWAPI version 4.4.0

Visual Studio 2017 10.0.17763.0, Windows 10

`Severity	Code	Description	Project	File	Line	Suppression State Error	LNK1181	cannot open input file 'BWAPI.lib' CommandCenter	C:\SCBW\masterBot\vs\LINK	1`

`1>LINK : fatal error LNK1181: cannot open input file 'BWAPI.lib'
1>Done building project "CommandCenter.vcxproj" -- FAILED.`

**Comment by vlad-gregor** [comment]
Yeah, I've just encountered the same issue. However, I resolved it by building BWAPI (say in RELEASE mode) which will produce several *.lib files in the Release folder of BWAPI, namely:

- BWAPILIB.lib
- BWAPICore.lib
- ...

And then I set the "Library Directory" setting of "VC++ Directories" to that. In my case the folder is:
c:\Libraries\bwapi_440.src\bwapi\BWAPI\Release

However, I too have downloaded BWAPI 4.4.0 which seems to be incompatible with this version of CommandCenter, because I actually get a compilation error:
Line 437 of "BuildingManager.cpp" [method: CCTilePosition BuildingManager::getBuildingLocation(const Building & b)] cannot build, because "Protoss is not a member of BWAPI::Race".

Any ideas?

**Comment by vlad-gregor** [comment]
I tried to create a BWAPI.lib by changing a property of 'BWAPI' project.
Went to project properties, General -> Configuration Type -> Static library (.lib).
That built fine and produced the BWAPI.lib.

However, the CommandCenter still does not build. This time linker showed 157 errors complaining about mismatches mostly in BWAPIClient.lib:
![Uploading Errors.png…]()

Any ideas?
